src/Organizations/OrganizationForm/OrganizationSummaryForm/FieldCode/FieldCode.js

```diff
@@ -16,15 +20,47 @@ const FieldCode = ({ orgId, mutator }) => {
   // eslint-disable-next-line react-hooks/exhaustive-deps
   [orgId]);
 
+  const { change } = useForm();
+  const stripes = useStripes();
+
+  const { settings } = useSettings([CONFIG_NAME]);
+  let vendorCodeSetting = 'useTextField';
+
+  if (stripes.hasInterface('servint')) {
+    vendorCodeSetting = settings?.find(sett => sett?.configName === CONFIG_NAME)?.parsedSettings?.vendorGeneratorSetting ?? 'useTextField';
+  }
+
   return (
-    <Field
-      component={TextField}
-      fullWidth
-      label={<FormattedMessage id="ui-organizations.summary.code" />}
-      name="code"
-      required
-      validate={validate}
-    />
+    <Row>
+      <Col xs={12}>
+        <Field
+          component={TextField}
+          disabled={vendorCodeSetting === 'useGenerator'}
+          fullWidth
+          label={<FormattedMessage id="ui-organizations.summary.code" />}
+          name="code"
+          required
+          validate={validate}
+        />
+      </Col>
+      {(
+        vendorCodeSetting === 'useGenerator' ||
```

Comment:
plz don't use magic words, additionally IMO you can return boolean values from hook (see previous comment)